### ISSUE TITLE
Not query new data when write in quick speed

### ISSUE THREAD

**Issue by woddle** [questdb/c-questdb-client#37]
When I flush data in quick speed about 10ms, then query in WebUI, but get not newest data;

**Comment by amunra** [comment]
This is expected and by design. The protocol is async and doesn't block until data is ready to be queried before accepting new data. Doing otherwise would severly impact ingestion performance.

Instead, if you want to query specific data, retry the query until that data is available with a small delay in between: this is generally the technique to use in test cases. 


A bit of background:

Internally the database processes rows in batches and there's always a small delay. The database also spends some time sorting the data into timestamp order. 


Optimising ingest-to-query latency:

If you want to minimise this delay try to send data as sorted as you can (by timestamp) without unduly complicating your own client code. Slightly unsorted data (events a few seconds out of order) generally has little to no performance impact, but ingesting data with very old events (say a few hours in the past) would end up impacting a large amount of data on disk due to sorting.

In short: If you bulk-insert pre-existing data, pre-sort it. If you're ingesting events had they happen don't bother sorting.

**Comment by amunra** [comment]
I forgot to mention that if your data doesn't appear at all, then there's probably an error in the ingestion.

On error, for example if you insert a string into a float column the database will:
* Log the error. 
* Disconnect. 

Always check the log files if you're unsure.

**Comment by amunra** [comment]
I will close this issue now, but feel free to re-open it if my response hadn't helped you.

If you're still having issues, please include as much debugging information as you can.
